### PR TITLE
fix: disable sign out button action, site settings when offline

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Disable sign out button action when offline.

### Related Issues
Addresses #2376, #2371.

### CODE REVIEW

File: dev-client/src/components/buttons/DeleteButton.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ export default function DeleteButton({label, ...props}: Props) {
       p={0}
       pt="10px"
       startIcon={<Icon name="delete" color="error.main" size="md" mr={2} />}
+      isDisabled={props?.disabled ? props.disabled : undefined}
```
Comment (knipec) [inline dev-client/src/components/buttons/DeleteButton.tsx:39]: Suggestion/question: This is equivalent to just `props?.disabled`, right?

File: dev-client/src/components/buttons/DeleteButton.tsx
Comment (knipec) [inline dev-client/src/components/buttons/DeleteButton.tsx:1]: Issue: It looks like the button is still red, should be grayed out
Screenshot 2024-12-06 at 3 40 03 PM

File: dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx [hunk 2]
```diff
@@ -16,28 +16,38 @@
...
-}
+};
+
+export const SignOutItem = () => {
+  const isDisabled = useIsOffline();
+
+  return isDisabled ? (
+    <SignOutMenuItem disabled={isDisabled} />
+  ) : (
+    <SignOutModal trigger={onOpen => <SignOutMenuItem onPress={onOpen} />} />
```
Comment (knipec) [inline dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx:51]: Note: Hmm the modal/trigger hierarchy here makes this read particularly awkward. We may want to rethink our pattern here if we're going to have to end up doing this in a lot of different places.

Comment (paulschreiber) [reply dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx:51]: I couldn't find a way to disable the trigger itself. I tried disabling onPress, but that didn't work.

File: dev-client/src/components/buttons/DeleteButton.tsx (hunk 1 above)
Comment (paulschreiber) [reply dev-client/src/components/buttons/DeleteButton.tsx:39]: That results in a TypeScript error:
```
Type 'boolean | null | undefined' is not assignable to type 'boolean | undefined'.
  Type 'null' is not assignable to type 'boolean | undefined'.ts(2322)
```

File: dev-client/src/components/buttons/DeleteButton.tsx
Comment (paulschreiber) [reply dev-client/src/components/buttons/DeleteButton.tsx:1]: Fixed. Had to manually add disabled styles, even with `isDisabled={true}`.

File: dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx (hunk 2 above)
Comment (paulschreiber) [reply dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx:51]: @tm-ruxandra any suggestions?

Comment (tm-ruxandra) [reply dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx:51]: @paulschreiber There's probably a better solution here but it requires re-working how our "trigger" system works. This is fine for now.

**Review by tm-ruxandra — APPROVED** [review]

**Pull request merged**